lib/experimental/colortext.nim

```diff
@@ -0,0 +1,765 @@
+import std/[terminal, unicode, sequtils, strformat, strutils]
+
+type
+  TermColor8Bit* = distinct uint8
+  TermColor8BitBg* = distinct TermColor8Bit
+  TermColor8BitFg* = distinct TermColor8Bit
+
+
+func col8*(color: uint8): TermColor8Bit = TermColor8Bit(color)
+func bg*(color: TermColor8Bit): TermColor8BitBg = TermColor8BitBg(color)
+func fg*(color: TermColor8Bit): TermColor8BitFg = TermColor8BitFg(color)
+
+func `==`*(a, b: TermColor8Bit): bool = a.uint8 == b.uint8
+
+
+type
+  ColStyle* = object
+    case use8Bit*: bool
+      of true:
+        fg8*: TermColor8Bit
+        bg8*: TermColor8Bit
+
+      of false:
+        fg*: ForegroundColor
+        bg*: BackgroundColor
```

Comment:
8 bit colors are usually used to save storage space, but with this condition, no storage space will be saved at all.

Reply:
Because I'm not aiming to save storage space here